gpio_test.go

```diff
@@ -0,0 +1,158 @@
+package gpio
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// valudate and extract test setup
+func getGpioSetup(t *testing.T) (gpiodev string, gpioin uint32, gpioout uint32) {
+	if len(os.Getenv("GPIO_DEV_PATH")) == 0 {
+		t.Skip("Please set GPIO_DEV_PATH, GPIO_TEST_PIN to run tests ")
+		return "", 0, 0
+	}
+	if len(os.Getenv("GPIO_TEST_PIN")) == 0 {
+		t.Skip("Please set GPIO_DEV_PATH, GPIO_TEST_PIN to run tests ")
+		return "",0, 0
+	}
+	testPinIn, err := strconv.Atoi(os.Getenv("GPIO_TEST_PIN"))
+	if (testPinIn < 1 || err != nil) {
+		t.Errorf("IO pin is below 1 or unparsable: %d|%s", testPinIn, err)
+		return "",0, 0
+	}
+	if len(os.Getenv("GPIO_TEST_PIN_LOOP")) == 0 {
+		t.Skip("Please set GPIO_TEST_PIN_LOOP and connect it with GPIO_TEST_PIN physically to run tests")
+	}
+	testPinOut, err := strconv.Atoi(os.Getenv("GPIO_TEST_PIN_LOOP"))
+	if (testPinOut < 1 || err != nil) {
+		t.Errorf("IO loop pin is below 1 or unparsable: %d|%s", testPinOut, err)
+		testPinOut = 0
+	}
+	if testPinIn == testPinOut {
+		t.Errorf("test pins can't be same %d|%d", testPinIn, testPinOut)
+	}
+
+	return os.Getenv("GPIO_DEV_PATH"),uint32(testPinIn), uint32(testPinOut)
+
+}
+
+
+
+func TestGPIO(t *testing.T) {
+	a := assert.New(t)
+	testDevice, testPin, testPinLoop := getGpioSetup(t)
+	fmt.Printf("dev: %s in: %d, out: %d\n", testDevice,testPin,testPinLoop)
+	// no test pin provided, skip
+	if testPin == 0 { return }
+	chiper, err := Open(testDevice, "go-test")
+	if !a.Nil(err,"Open should succeed") { return }
```

Comment:
Please make `assert.Nil(t, err)` or at least `assert:=assert.New(t)`
current `a.Nil` is not immediately easy to understand.